perfkitbenchmarker/linux_benchmarks/object_storage_service_benchmark.py

```diff
@@ -346,48 +355,105 @@ def _ProcessMultiStreamResults(raw_result, operation, sizes,
     sizes: the object sizes used in the benchmark, in bytes.
     results: a list to append Sample objects to.
     metadata: dict. Base sample metadata
-
   """
 
+  num_streams = FLAGS.object_storage_multistream_num_streams
+
   if metadata is None:
     metadata = {}
-
-  records = json.loads(raw_result)
-  metric_name = 'Multi-stream %s latency' % operation
-
-  metadata = metadata.copy()
-  metadata['num_streams'] = FLAGS.object_storage_multistream_num_streams
+  metadata['num_streams'] = num_streams
   metadata['objects_per_stream'] = (
       FLAGS.object_storage_multistream_objects_per_stream)
 
-  overall_metadata = metadata.copy()
+  records_json = json.loads(raw_result)
+  records = pd.DataFrame(records_json)
+
+  all_streams_start, all_streams_duration = analyze.AllStreamsInterval(
```

Comment:
Also, I am trying to come up with a better name than "all_streams_start", "all_streams_duration". These names actually gives people a first impression: the start time or the duration of all streams, which means earliest start time and the entire duration covers all streams, which is exactly the opposite of what they actually mean.

all_streams_start is actually the latest start time of all streams, right?
more descriptives

how about these two? - 
"latest_start_time_of_all_streams"
"duration_when_all_streams_active"